src/js/editor/editor.js

```diff
@@ -673,10 +686,32 @@ class Editor {
    *
    * @param {Function} callback Called with an instance of
    *        {@link PostEditor} as its argument.
+   * @param {Number|undefined} actionType provides the type of the current
+   *                                      action, but only if it's a "runnable"
+   *                                      type.
    * @return {Mixed} The return value of `callback`.
    * @public
    */
-  run(callback) {
+  run(callback, actionType) {
+
+    // Decides whether or not this current action is in a run of previous
+    // actions which dictates whether or not it overwrites the last item on the
+    // undo queue.
+    let isInARun = actionType && this._undoRunLastState === actionType;
+    this._undoRunLastState = actionType;
+
+    if(this._undoRunTimeout) {
+      clearTimeout(this._undoRunTimeout);
+      this._undoRunTimeout = null;
+    }
+
+    if(isInARun) {
+        this._undoRunTimeout = setTimeout(() => {
+          this._undoRunTimeout = null;
+          this._undoRunLastState = null;
+        },
+        UNDO_RUN_BREAK_TIMEOUT_LENGTH);
```

Comment:
`this._undoRunTimeout` probably needs to be explicitly cleared in `editor#destroy` as well (if it is present)